### PR TITLE
fix: stableswap joinPoolSharesInternal

### PR DESCRIPTION
**Pull request by catShaark**
Closes: #XXX

## What is the purpose of the change

 Stable swap `joinPoolSharesInternal` doesn't `updatePoolForJoin` for the case of single asset join  

## Brief Changelog

I change `joinPoolSharesInternal` so that it `updatePoolForJoin` for the case of single asset join

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes?  no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not documented

### CODE REVIEW

File: x/gamm/pool-models/stableswap/amm.go [hunk 1]
```diff
@@ -378,7 +378,11 @@ func (pa *Pool) calcSingleAssetJoinShares(tokenIn sdk.Coin, swapFee sdk.Dec) (sd
 func (pa *Pool) joinPoolSharesInternal(ctx sdk.Context, tokensIn sdk.Coins, swapFee sdk.Dec) (numShares sdk.Int, newLiquidity sdk.Coins, err error) {
 	if len(tokensIn) == 1 {
 		numShares, err = pa.calcSingleAssetJoinShares(tokensIn[0], swapFee)
+		if err != nil {
+			return sdk.ZeroInt(), sdk.NewCoins(), err
+		}
```
Comment (alexanderbez) [inline x/gamm/pool-models/stableswap/amm.go:383]:
```suggestion
		}

```

File: x/gamm/pool-models/stableswap/amm.go [hunk 2]
```diff
@@ -378,7 +378,11 @@ func (pa *Pool) calcSingleAssetJoinShares(tokenIn sdk.Coin, swapFee sdk.Dec) (sd
...
 		newLiquidity = tokensIn
+		pa.updatePoolForJoin(tokensIn, numShares)
```
Comment (alexanderbez) [inline x/gamm/pool-models/stableswap/amm.go:385]:
```suggestion
		pa.updatePoolForJoin(tokensIn, numShares)

```

**Review by alexanderbez — COMMENTED** [review]
Shouldn't there be a unit test(s) that is failing due to this? If not, we need to add one.

**Review by p0mvn — APPROVED** [review]

**Comment by p0mvn** [comment]
Seems that @catShaark is working on unit tests in: https://github.com/osmosis-labs/osmosis/issues/1726

**Comment by catShaark** [comment]
> Seems that @catShaark is working on unit tests in: #1726

should I add the unit test in this PR or in another PR ?

**Comment by p0mvn** [comment]
> > Seems that @catShaark is working on unit tests in: #1726
> 
> should I add the unit test in this PR or in another PR ?

I think in this PR would be best if that's okay with you. It would allow us to compare the change to what unit tests try to achieve.

**Comment by catShaark** [comment]
> > > Seems that @catShaark is working on unit tests in: #1726
> > 
> > 
> > should I add the unit test in this PR or in another PR ?
> 
> I think in this PR would be best if that's okay with you. It would allow us to compare the change to what unit tests try to achieve.

I'll include the test in this PR

**Pull request closed without merging**